### PR TITLE
Modified rebuild cache logic when truncate cache

### LINKED ISSUE #25: no need to read the whole idx file just to restore the latest entry in cache
init_cache_fro_file is not necessary, it should be enough to read only the last entry from file

```
-spec truncate_cache(cache(), filename(), logid()) -> cache().
truncate_cache(Tid, NewSegIdxFile, LogId) ->
  Ms = ets:fun2ms(fun(?ENTRY(LogIdX, _, _, _)) -> LogIdX >= LogId end),
  _  = ets:select_delete(Tid, Ms),
  %%   To make sure that the latest entry is always in cache.
  ok = init_cache_from_file(NewSegIdxFile, Tid),
  Tid.
```

### PR DESCRIPTION
**Pull request by redink**
closes #25

### CODE REVIEW

File: src/gululog_idx.erl [hunk 1]
```diff
@@ -528,6 +528,30 @@ init_cache_from_file(_Version = 1, Tid, SegId, Fd) ->
...
+restore_cache_from_latest_entry(_Version = 1, Tid, Fd, SegId, ChunkBin) ->
+  case file:read(Fd, ?FILE_ENTRY_BITS_V1 * ?FILE_READ_CHUNK) of
+    eof ->
+      [ ets:insert(Tid, from_file_entry(1, SegId, <<E:?FILE_ENTRY_BITS_V1>>))
+        || <<E:?FILE_ENTRY_BITS_V1>> <= ChunkBin],
+      ok;
+    {ok, TmpChunkBin} ->
+      restore_cache_from_latest_entry(1, Tid, Fd, SegId, TmpChunkBin)
+  end.
+
```
Comment (redink) [inline src/gululog_idx.erl]: To get the latest entry from index file:
1, read the index file and recording the last time `ChunkBin`;
2, when read return `eof`, inert `ChunkBin` into cache.

File: src/gululog_idx.erl [hunk 2]
```diff
@@ -340,7 +340,7 @@ truncate_cache(Tid, NewSegIdxFile, LogId) ->
   Ms = ets:fun2ms(fun(?ENTRY(LogIdX, _, _, _)) -> LogIdX >= LogId end),
   _  = ets:select_delete(Tid, Ms),
   %%   To make sure that the latest entry is always in cache.
-  ok = init_cache_from_file(NewSegIdxFile, Tid),
+  ok = restore_cache_from_latest_entry(NewSegIdxFile, Tid),
```
Comment (zmshi) [inline src/gululog_idx.erl]: perhaps a better name: restore_latest_cache_entry_from_file

File: src/gululog_idx.erl (hunk 1 above)
Comment (zmshi) [inline src/gululog_idx.erl]: There is a quite simpler logic:

```
{ok, EofPosition} = file:position(Fd, eof),
EntrySize = file_entry_bytes(Version),
{ok, Bin} = file:pread(Fd, EofPosition - EntrySize, EntrySize)
...
```

Comment (redink) [inline src/gululog_idx.erl]: Got it . :+1:

**Pull request merged**